### PR TITLE
Upgrate kind, Use github runners for CI

### PR DESCRIPTION
**Pull request by dhschall**
This should move all CI to github hosted runners to avoid failing test.
We also upgrade the kind modules

### CODE REVIEW

File: .github/workflows/e2e-aes.yml [hunk 1]
```diff
@@ -111,12 +110,13 @@ jobs:
     name: Test Knative Deployment
     needs: build-and-push
     env:
-      GOCACHE: /root/tmp/gocache
-      GOPATH: /root/tmp/gopath
-    runs-on: [self-hosted, stock-knative]
+      KIND_VERSION: v0.14.0
+      K8S_VERSION: v1.22.10
```
Comment (ustiugov) [inline .github/workflows/e2e-aes.yml:114]: we use 1.23!

File: .github/workflows/e2e-aes.yml [hunk 2]
```diff
@@ -140,24 +140,58 @@ jobs:
...
+          KNATIVE_DOMAIN=$INGRESS_HOST.nip.io
+          kubectl patch configmap -n knative-serving config-domain -p "{\"data\": {\"$KNATIVE_DOMAIN\": \"\"}}"
+
+
+      # - name: Kubernetes KinD Cluster
+      #   uses: container-tools/kind-action@v1
+      #   with:
+      #     version: v0.14.0
+      #     kubectl_version: v1.22.10
+      #     knative_serving: v1.2.2
```
Comment (ustiugov) [inline .github/workflows/e2e-aes.yml:173]: we use knative 1.4 in vHive main branch

File: runner/scripts/02-kourier.sh [hunk 3]
```diff
@@ -3,14 +3,15 @@
 set -eo pipefail
 set -u
 
-KNATIVE_NET_KOURIER_VERSION=${KNATIVE_NET_KOURIER_VERSION:-0.23.0}
+KNATIVE_NET_KOURIER_VERSION=${KNATIVE_NET_KOURIER_VERSION:-1.2.0}
```
Comment (ustiugov) [inline runner/scripts/02-kourier.sh:6]:
```suggestion
KNATIVE_NET_KOURIER_VERSION=${KNATIVE_NET_KOURIER_VERSION:-1.4.0}
```

File: runner/scripts/04-eventing.sh [hunk 4]
```diff
@@ -3,39 +3,43 @@
 set -eo pipefail
```
Comment (ustiugov) [inline runner/scripts/04-eventing.sh:3]: I don't think we should test eventing at this point. let's just get serving CI running

File: .github/workflows/e2e-auth.yml [hunk 5]
```diff
@@ -137,36 +135,57 @@ jobs:
...
+      - name: Install Serving
+        run: bash ./runner/scripts/02-serving.sh
 
+      - name: Install Kourier
+        run: bash ./runner/scripts/02-kourier.sh
+
+      - name: Setup domain
+        run: |
+          INGRESS_HOST="127.0.0.1"
+          KNATIVE_DOMAIN=$INGRESS_HOST.nip.io
```
Comment (ustiugov) [inline .github/workflows/e2e-auth.yml:153]: this is a bug! and it appears elsewhere in this PR

there is no nip.io anymore, we use sslip.io instead
https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/#configure-dns

**Review by ustiugov — COMMENTED** [review]
just made some points but didn't have time to make a proper review.

**Comment by ustiugov** [comment]
so you did make knative serving work on GH-hosted runners! bravo, guys!

File: runner/scripts/04-eventing.sh (hunk 4 above)
Comment (dhschall) [reply runner/scripts/04-eventing.sh]: Removed the file

File: .github/workflows/e2e-auth.yml (hunk 5 above)
Comment (dhschall) [reply .github/workflows/e2e-auth.yml]: Thanks!!
Good hint

**Comment by dhschall** [comment]
Ok so now I think its good for your proper review.
- I fixed now all things in the knative testing. They use now the github hosted runners.
- I applied the changes to the standalone functions (AES, Auth, Fibonacci, Hotel-app and online shop)
  - Beside online shop all of them seem to work. 
  - Online shop fails in the build and does not belong to this issue but I created a new issue for it.
Other workflows are not adjusted yet. We still need to do this. 
However, imho we should not fix failing benchmark but instead create separate issues for failing once.

File: .github/workflows/e2e-aes.yml [hunk 6]
```diff
@@ -93,10 +95,9 @@ jobs:
           lfs: "true"
 
       - name: start docker-compose benchmark
-        working-directory: benchmarks/aes/yamls/docker-compose
         run: |
-          docker-compose -f ${{ matrix.service }}.yaml pull
-          docker-compose -f ${{ matrix.service }}.yaml up &> log_file &
+          docker-compose -f ${{ env.YAML_DIR }}/dc-${{ matrix.service }}.yaml pull
```
Comment (ustiugov) [inline .github/workflows/e2e-aes.yml:99]: If you renamed the benchmarks, please rename these as well

File: .github/workflows/e2e-hotel.yml [hunk 7]
```diff
@@ -134,37 +136,66 @@ jobs:
...
+
+          ./test-client --addr $url:$NODEPORT --name "Example text for Hotel App CI"
 
       - name: Print logs
         if: ${{ always() }}
         run: |
           set -x
-          container_list=$(kn service export ${{ matrix.service }} -o jsonpath="{.spec.service.spec.template.spec.containers[*].name}")
-          for container_name in $container_list
+          # container_list=$(kubectl get pods -n default -o jsonpath="{.items[*].spec.containers[*].name}")
```
Comment (ustiugov) [inline .github/workflows/e2e-hotel.yml:190]: Remove comments

File: runner/scripts/01-kind.sh [hunk 8]
```diff
@@ -3,44 +3,32 @@
...
   kind delete cluster --name ${CLUSTER_NAME}
-elif [ "$REPLY" == "N" ] || [ "$REPLY" == "n" ] || [ -z "$REPLY" ]; then
-  echo "Installation skipped"
-  exit 0
 fi
 
+echo "Using image ${KIND_BASE}:${K8S_VERSION}"
 KIND_CLUSTER=$(mktemp)
-cat <<EOF | kind create cluster --name ${CLUSTER_NAME} --config=-
+cat <<EOF | kind create cluster --name ${CLUSTER_NAME} --wait 120s --wait 120s --config=-
```
Comment (ustiugov) [inline runner/scripts/01-kind.sh:24]: Duplicate option

**Review by ustiugov — CHANGES_REQUESTED** [review]
Looks great. Some minor issues. 

Also please raise new Issues for the workloads that don't have the support of gh runners yet and mention them here. Then, please go ahead with the merge, no need for another review by me.

**Comment by dhschall** [comment]
> and 

Thanks Dmitrii! I'll finalise this once I'm back next Friday.

**Comment by dhschall** [comment]
This PR closes #237.
The new workflow is tested on several benchmarks and works. It needs to be applied to the remining once here #248.

**Pull request merged**